bundles/org.eclipse.ui.navigator/src/org/eclipse/ui/internal/navigator/actions/CommonActionProviderDescriptor.java

```diff
@@ -231,17 +234,44 @@ public boolean isEnabledFor(IStructuredSelection aStructuredSelection) {
 			}
 		} else {
 			IEvaluationContext parentContext = NavigatorPlugin.getApplicationContext();
+			String multiproject = configurationElement.getAttribute(MULTI_PROJECT_ENABLE);
+			boolean retValue = false;
 			for (Object element : aStructuredSelection) {
 				IEvaluationContext context = new EvaluationContext(parentContext, element);
 				context.setAllowPluginActivation(true);
 				if (NavigatorPlugin.safeEvaluate(enablement, context) != EvaluationResult.TRUE) {
-					return false;
+					boolean isProject = isProject(element.getClass());
+					if (!isProject || multiproject == null || !multiproject.equals("any")) { //$NON-NLS-1$
+						return false;
+					}
+				} else {
+					retValue = true;
```

Comment:
as retvalue could never change here, why not return immediately?

Reply:
Because this else statement is also hit in the normal case where there is no multi-project enable "any".  There may be other elements in the selection to check in the loop which could return false.